include/ocpp/common/message_queue.hpp

```diff
@@ -661,16 +701,16 @@ template <typename M> class MessageQueue {
     }
 
     /// \brief Sends a new \p call_result message over the websocket
-    template <class T> void push(CallResult<T> call_result) {
+    void push_call_result(const json& call_result_json, const MessageId& unique_id) override {
```

Comment:
so we don't need this to be a template any more because we just separate out the structure?

Reply:
The template part is now in the interface, which delegates to the derived class implementation, which does not need a template.

Basically, in C++ you cannot have a pure virtual function that is templated, yet we wanted a MessageQueueInterface to allow for the ability to mock it in tests. So some of the logic was moved into the interface template functions in order to create and pass the non-templated parameters to the derived implementation.